From 32d07b2d16c4e71a98d1b8d0163ef525b6e7ed19 Mon Sep 17 00:00:00 2001
From: Mikolaj Gralczyk <mgralczyk@anaconda.com>
Date: Tue, 31 Mar 2026 11:47:43 +0200
Subject: [PATCH] Create 1D list instead of 2D

This fixes Windows issue reflected by 'test_write_read_gsb'.
Caused by Fortran convention mismatch between platforms (works on Unix).
---
 src/pygeostat/data/iotools.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pygeostat/data/iotools.py b/src/pygeostat/data/iotools.py
index 39040f3..1285640 100644
--- a/src/pygeostat/data/iotools.py
+++ b/src/pygeostat/data/iotools.py
@@ -561,7 +561,7 @@ def write_gsb(data, flname, tvar=None, nreals=1, variables=None, griddef=None, f
     # Character conversion of strings
     cvariables = []
     for varname in [var.ljust(64) for var in variables]:
-        cvariables.append([v for v in varname])
+        cvariables.extend([v for v in varname])
 
     tmin = Parameters.get('data.tmin', None)
     if tmin is None:
-- 
2.50.1 (Apple Git-155)

